doc/tutorials/filesets.rst

```diff
@@ -27,3 +28,243 @@ It's assumed that you're already familiar with the :doc:`dataset basics
+          SERIES: Modality=CT, SeriesNumber=2
+            IMAGE: SOPInstanceUID=1.3.6.1.4.1.5962.1.1.0.0.0.1196530851.28319.0.93
+            IMAGE: SOPInstanceUID=1.3.6.1.4.1.5962.1.1.0.0.0.1196530851.28319.0.94
+            IMAGE: SOPInstanceUID=1.3.6.1.4.1.5962.1.1.0.0.0.1196530851.28319.0.95
+            IMAGE: SOPInstanceUID=1.3.6.1.4.1.5962.1.1.0.0.0.1196530851.28319.0.96
+      PATIENT: PatientID=98890234, PatientName=Doe^Peter
+       STUDY: StudyDate=20010101, StudyTime=000000
+          SERIES: Modality=CT, SeriesNumber=4
+            IMAGE: SOPInstanceUID=1.3.6.1.4.1.5962.1.1.0.0.0.1194734704.16302.0.3
+            IMAGE: SOPInstanceUID=1.3.6.1.4.1.5962.1.1.0.0.0.1194734704.16302.0.5
+          ...
+          SERIES: Modality=MR, SeriesNumber=700
+            IMAGE: SOPInstanceUID=1.3.6.1.4.1.5962.1.1.0.0.0.1196533885.18148.0.121
+            IMAGE: SOPInstanceUID=1.3.6.1.4.1.5962.1.1.0.0.0.1196533885.18148.0.120
+            IMAGE: SOPInstanceUID=1.3.6.1.4.1.5962.1.1.0.0.0.1196533885.18148.0.122
+            IMAGE: SOPInstanceUID=1.3.6.1.4.1.5962.1.1.0.0.0.1196533885.18148.0.119
+            IMAGE: SOPInstanceUID=1.3.6.1.4.1.5962.1.1.0.0.0.1196533885.18148.0.123
+            IMAGE: SOPInstanceUID=1.3.6.1.4.1.5962.1.1.0.0.0.1196533885.18148.0.125
+            IMAGE: SOPInstanceUID=1.3.6.1.4.1.5962.1.1.0.0.0.1196533885.18148.0.124
+
+Rather than representing the File-set as a tree model, the
+:class:`~pydicom.dicomdir.FileSet` class treats it as a flat
+collection of SOP Instances, abstracting away the need to dig down into the
+hierarchy (although that's still possible). For example, iterating over the
+:class:`~pydicom.dicomdir.FileSet` yields a
+:class:`~pydicom.dicomdir.FileInstance` object for each of the managed SOP
+Instances.
+
+.. code-block:: python
+
+    >>> for instance in fs:
+    ...     print(instance.PatientName)
+    ...     break
+    Doe^Archibald
+
+The managed instances in the File-set can be searched and then loaded to return
+a :class:`~pydicom.dataset.Dataset`:
+
+.. code-block:: python
+
+    >>> fs.find_values("PatientID")
+    ['77654033', '98890234']
+    >>> for instance in fs.find(PatientID='77654033'):
+    ...     ds = instance.load()
+    ...     print(ds.PhotometricInterpretation)
+    MONOCHROME1
+    MONOCHROME1
+    MONOCHROME1
+    MONOCHROME2
+    MONOCHROME2
+    MONOCHROME2
+    MONOCHROME2
+
+By default, both :meth:`~pydicom.dicomdir.FileSet.find` and
+:meth:`~pydicom.dicomdir.FileSet.find_values` only search the elements within
+the directory records of the DICOMDIR file. You can search for any
+element within the actual stored instances by using the *load*
+keyword parameter:
+
+.. code-block:: python
+
+    >>> fs.find_values("PhotometricInterpretation")
+    []
+    >>> fs.find_values("PhotometricInterpretation", load=True)
+    ['MONOCHROME1', 'MONOCHROME2']
+    >>> len(fs.find(PatientID='77654033', PhotometricInterpretation='MONOCHROME1'))
```

Comment:
Just starting to read through the docs to understand this PR ... the 'loading' concept was initially a little unclear - IIUC, it is to get all information in the dataset, as opposed to only what is recorded in the DICOMDIR?  It seems strange to allow a search on items that for sure won't be in a non-loaded instance.

Edit: I do see it described well in the find_values() function, but my comment re searching still stands.

Reply:
I'll be honest, I put that there for me 😉, I often have to transfer a series or two for a patient so it can be decompressed before sending it on to the planning systems. Usually the only information I'll have is the *Series Description*, which isn't in the DICOMDIR. It does default to not loading the instances, as it could be quite slow.

Reply:
I think maybe I wasn't clear - I like the idea of being able to search anything, I just don't see the point of having a `load` flag as an argument - if the keyword is not available in the DICOMDIR records, then to me it should automatically be searched by loading, else why call the function at all, when you know you are going to get an empty return value?  At the very least, IMO the function should raise an error to show that the keyword is not available in the DICOMDIR records, and note that the user should set the `load` flag.

Reply:
> At the very least, IMO the function should raise an error to show that the keyword is not available in the DICOMDIR records, and note that the user should set the load flag.

Yeah, this makes sense. I'm worried about the potential time it would take to search all the stored instances if it searches them by default.